core/server/api/canary/utils/serializers/output/utils/clean.js

```diff
@@ -95,6 +95,10 @@ const post = (attrs, frame) => {
         if (attrs.og_description === '') {
             attrs.og_description = null;
         }
+        // NOTE: the visibility column has to be always present in Content API response to perform content gating
+        if (frame.options.columns && frame.options.columns.includes('visibility') && !frame.original.query.fields.includes('visibility')) {
```

Comment:
A nicer way would be having `columns` in the `original` object as well, but don't think it's worth changing the internal structure for just this one case. Additionally, visibility can only be passed in through `fields` which prevents from missing this field elsewhere (e.g. plaintext can be defined in both `fields` and `formats`). 

Didn't have to implement symmetric change in API v2 as the field is always deleted there.